File: src/api/schemas/datasets.py
"""数据集接口 Pydantic 模型"""

from datetime import datetime

from pydantic import BaseModel, Field


class DatasetCreateRequest(BaseModel):
    name: str = Field(..., min_length=1, max_length=256)
    description: str | None = None


class DatasetUpdateRequest(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=256)
    description: str | None = None


class DatasetResponse(BaseModel):
    dataset_id: str
    name: str
    description: str | None = None
    doc_count: int
    created_at: datetime | None = None
    updated_at: datetime | None = None


class DatasetListResponse(BaseModel):
    total: int
    page: int
    size: int
    items: list[DatasetResponse]

File: web/src/api/datasets.ts
import request from './request'

export interface DatasetCreateRequest {
  name: string
  description?: string
}

export interface DatasetUpdateRequest {
  name?: string
  description?: string
}

export interface DatasetResponse {
  dataset_id: string
  name: string
  description: string | null
  doc_count: number
  created_at: string | null
  updated_at: string | null
}

export interface DatasetListResponse {
  total: number
  page: number
  size: number
  items: DatasetResponse[]
}

export const createDataset = (data: DatasetCreateRequest) =>
  request.post<any, DatasetResponse>('/datasets', data)

export const listDatasets = (params: { page?: number; size?: number }) =>
  request.get<any, DatasetListResponse>('/datasets', { params })

export const getDataset = (datasetId: string) =>
  request.get<any, DatasetResponse>(`/datasets/${datasetId}`)

export const updateDataset = (datasetId: string, data: DatasetUpdateRequest) =>
  request.patch<any, DatasetResponse>(`/datasets/${datasetId}`, data)

export const deleteDataset = (datasetId: string, force = false) =>
  request.delete<any, { message: string; dataset_id: string }>(`/datasets/${datasetId}`, { params: { force } })
